bittensor/subtensor.py

```diff
@@ -912,8 +912,8 @@ def commit_weights(
         wallet: "bittensor.wallet",
         netuid: int,
         salt: List[int],
-        uids: Union[NDArray[np.int64], list],
-        weights: Union[NDArray[np.float32], list],
+        uids: Union[NDArray[np.int64]],
+        weights: Union[NDArray[np.float32]],
```

Comment:
Why the use of `Union` here? The typing here implies `uids` will only be an NDArray of int64, and weight only an NDArray of float32.

Reply:
Great catch, thanks